mibidata/mibi_image.py

```diff
@@ -185,6 +185,70 @@ def __init__(self, data, channels, **kwargs):
             setattr(self, k, v)
             self._user_defined_attributes.append(k)
 
+    def add_attr(self, **kwargs):
+        """Adds user-defined metadata key-value pairs as attributes to
+           the class instance in use. If attribute already exists
+           for the current instance, raises an error.
+
+        Args:
+            kwargs: A mapping of arguments for a user to add multiple
+                    attributes with their respecitve
+                    values.
+        Raises:
+            ValueError: Raised if attempts to set an attribute that is already
+            defined for this instance.
+
+        """
+        already_defined = [attr for attr in kwargs if hasattr(self, attr)]
+        if already_defined:
+            if len(already_defined) == 1:
+                already_def_error = (f'{already_defined[0]} is already '
+                                     f'an attribute of this instance.')
+            else:
+                already_def_error = (f'{", ".join(already_defined)} are '
+                                     f'already attributes for this instance.')
+            raise ValueError(already_def_error)
+        for key, value in kwargs.items():
+            self._user_defined_attributes.append(key)
+            setattr(self, key, value)
+
+
+    def remove_attr(self, attributes):
+        """Removes user-defined attributes from the class instance in use.
+
+        Args:
+            attributes: A list of user-defined attributes for deletion.
```

Comment:
The only suggestion left I would make is allow attributes to be a string or a list of strings so that if a user is trying to remove a single attribute he/she doesn't have to pass it as a list of one item.